### PR TITLE
Rtc tests

### PR DESCRIPTION
**Pull request by NikhitaR-IFX**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
Enhanced RTC for testability and completing pending features. Details:

- Added RTC.irq(), RTC.alarm(), RTC.alarm_left(), RTC.cancel() and RTC.memory() functions
- Minor refactoring  in existing api's.
- Extended tests for coverage
- Modified docs to include latest api's

### CODE REVIEW

File: ports/psoc6/main.c [hunk 1]
```diff
@@ -186,6 +187,9 @@ void mpy_task(void *arg) {
     mod_i2c_deinit();
     mod_pwm_deinit();
     mod_spi_deinit();
+    // if (rtc_memory_write_enabled() == false) {
```
[the remark this answers is not in the document]
Comment (jaenrig-ifx) [reply ports/psoc6/main.c]: done!

File: ports/psoc6/machine_rtc.c [hunk 2]
```diff
@@ -58,42 +60,49 @@
...
-    return (mp_obj_t)&machine_rtc_obj;
+void mod_rtc_deinit() {
+    if (rtc_memory_write_enabled() == false) {
+        cyhal_rtc_free(&psoc6_rtc);
+    }
 }
 
 // Helper function to set/get datetime
-static mp_obj_t machine_rtc_datetime_helper(mp_uint_t n_args, const mp_obj_t *args) {
+static mp_obj_t machine_rtc_datetime_helper(mp_obj_t self_in, mp_uint_t n_args, const mp_obj_t *args) {
```
[the remark this answers is not in the document]
Comment (jaenrig-ifx) [reply ports/psoc6/machine_rtc.c]: done!

**Review by ramya-subramanyam — COMMENTED** [review]
very nice! just a minor feedback

File: tests/psoc6/rtc.py [hunk 3]
```diff
@@ -2,53 +2,108 @@
 from machine import RTC
 import time
 
-rtc = RTC()
-print(rtc)
+irq_counter = 0
+one_shot_alarm = 0
+periodic_alarm = 1
```
[the remark this answers is not in the document]
Comment (NikhitaR-IFX) [reply tests/psoc6/rtc.py]: Done now . Ideally all constants should be capitalized according PEP8 guidelines.

**Review by IFX-Anusha — APPROVED** [review]

**Review by ramya-subramanyam — APPROVED** [review]

**Review by jaenrig-ifx — APPROVED** [review]

**Pull request merged**